packages/odyssey-react-mui/src/labs/DataComponents/DataTable.tsx

```diff
@@ -0,0 +1,96 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { memo } from "react";
+
+import { DataView } from "./DataView";
+import { TableProps, UniversalProps } from "./componentTypes";
+
+export type DataTableProps = UniversalProps & TableProps;
+
+const DataTable = ({
+  bulkActionMenuItems,
+  columns,
+  currentPage,
+  emptyPlaceholder,
+  errorMessage,
+  filters,
+  getData,
+  hasChangeableDensity,
+  hasColumnResizing,
+  hasColumnVisibility,
+  initialDensity,
+  hasFilters,
+  hasPagination,
+  hasRowReordering,
+  hasRowSelection,
+  hasSearch,
+  hasSearchSubmitButton,
+  hasSorting,
+  isLoading,
+  isEmpty,
+  isNoResults,
+  isRowReorderingDisabled,
+  noResultsPlaceholder,
+  onChangeRowSelection,
+  paginationType,
+  renderDetailPanel,
+  resultsPerPage,
+  rowActionButtons,
+  rowActionMenuItems,
+  searchDelayTime,
+  totalRows,
+}: DataTableProps) => {
+  return (
+    <DataView
+      availableLayouts={["table"]}
+      bulkActionMenuItems={bulkActionMenuItems}
+      currentPage={currentPage}
+      emptyPlaceholder={emptyPlaceholder}
+      errorMessage={errorMessage}
+      filters={filters}
+      getData={getData}
+      hasFilters={hasFilters}
+      hasPagination={hasPagination}
+      hasRowReordering={hasRowReordering}
+      hasRowSelection={hasRowSelection}
+      hasSearch={hasSearch}
+      hasSearchSubmitButton={hasSearchSubmitButton}
+      isEmpty={isEmpty}
+      isLoading={isLoading}
+      isNoResults={isNoResults}
+      isRowReorderingDisabled={isRowReorderingDisabled}
+      noResultsPlaceholder={noResultsPlaceholder}
+      onChangeRowSelection={onChangeRowSelection}
+      paginationType={paginationType}
+      resultsPerPage={resultsPerPage}
+      searchDelayTime={searchDelayTime}
+      tableOptions={{
+        columns,
+        hasChangeableDensity,
+        hasColumnResizing,
+        hasColumnVisibility,
+        hasSorting,
+        initialDensity,
+        renderDetailPanel,
+        rowActionButtons,
+        rowActionMenuItems,
+      }}
```

Comment:
We should memoize `tableOptions`.

Reply:
By the way, wouldn't it be better to pass in a `DataTable` component with those props on it rather than passing them all as an object? Or is this how we did it in the past?

Reply:
I've memoized this now.

For clarity, the structure is like this:

`<DataView>` is the main component. It includes `<TableContent>` and `<StackContent>` sub-components, which render the particular views.

`<DataTable>` and `<DataStack>` are wrappers for convenience, so a consumer can implement a table-only `DataView` using the same API as the previous iteration of the DataTable.

Most of the props that go into `DataView` are universal (they apply to both stacks and tables), but some are specific to particular implementations. I went back and forth on how best to model these, and settled on objects for `tableOptions` and `stackOptions`, which provide some automatic namespacing. (`tableOptions.initialDensity` feels better to me than `tableInitialDensity`, especially in cases when the DataView doesn't include a table at all).